
DESARROLLO DE FUNCIONES

In [1]:
#Importacion de librerias a utilizar

import pandas as pd

In [2]:
#Ingesta del datafrmae generado luego del proceso de ETL
general_df =pd.read_csv('plataformas.csv')

-----------------------------------------------------------------------------
Consultas a generar:

Cantidad de veces que aparece una keyword en el título de peliculas/series, por plataforma

Cantidad de películas por plataforma con un puntaje mayor a XX en determinado año -- Ok

La segunda película con mayor score para una plataforma determinada, según el orden alfabético de los títulos.

Película que más duró según año, plataforma y tipo de duración ---- OK

Cantidad de series y películas por rating ---- OK

-----------------------------------------------------------------------------
Consulta 1: Cantidad de veces que aparece una keyword en el título de peliculas/series, por plataforma

In [3]:
def get_word_count(plataform, keyword):
    if plataform == "amazon":
        df_gwc = general_df[general_df['ID'].str.contains("a")]
    elif plataform == "dinsey":
        df_gwc = general_df[general_df['ID'].str.contains("d")]
    elif plataform == "hulu":
        df_gwc = general_df[general_df['ID'].str.contains("h")]
    elif plataform == "netflix":
        df_gwc = general_df[general_df['ID'].str.contains("n")]
    else:
        print("Try again")
    
    valor = df_gwc['title'].str.contains(keyword).value_counts()[True]

    return print("La palabra", keyword, "aparece", valor, "veces en la plataforma", plataform)

In [4]:
get_word_count("netflix", "love")

La palabra love aparece 196 veces en la plataforma netflix


La palabra love aparece 196 veces en la plataforma netflix
-----------------------------------------------------------------------------
Consulta 2: Cantidad de películas por plataforma con un puntaje mayor a XX en determinado año

In [5]:
def get_score_count(plataform, score, year):
    df5 = general_df[(general_df["score"] > score ) & (general_df["release_year"] == year)]  #Filtro de busqueda por Score y Año
    if plataform == "netflix":
        df_gsc = df5[(df5["type"] == "movie") & (df5["ID"].str.findall("n"))]                #Genero un DF general con filtros especificos segun busqueda
    elif plataform == "amazon":
        df_gsc = df5[(df5["type"] == "movie") & (df5["ID"].str.findall("a"))]
    elif plataform == "hulu":
        df_gsc = df5[(df5["type"] == "movie") & (df5["ID"].str.findall("h"))]
    elif plataform == "disney":
        df_gsc = df5[(df5["type"] == "movie") & (df5["ID"].str.findall("d"))]
    else:
        print("Try again")

    return print("Peliculas de", plataform, "del año", year, "con puntaje", score, ":", df_gsc.shape[0])
          

In [8]:

get_score_count('netflix', 85, 2010)

KeyError: 'score'

In [ ]:
get_word_count

In [ ]:
get_second_score

In [ ]:
get_longest

In [ ]:
get_rating_count